test/cmdlineTests/ir_compiler_subobjects/input.sol

```diff
@@ -0,0 +1,13 @@
+pragma solidity >=0.6.0;
+
+contract C {
+    uint public i;
```

Comment:
Can you make `C` empty so that it is clearer what happens? Maybe also use `--ir-optimized --optimize` as options?

Also, can you create a test that shows that the source of `C` is not included in `D` if `D` inherits from `C`, even though it is part of the contractDependencies?

Reply:
I added test with inheritance and changed to minimal test for this one. It has warning for unused variable. But shows the usage of the subObjects